src/orquestra/opt/optimizers/pso/continuous_pso_optimizer.py

```diff
@@ -0,0 +1,333 @@
+from typing import Callable, Optional, Sequence, Tuple, Union
+
+import numpy as np
+import scipy
+from scipy.optimize import OptimizeResult
+
+from orquestra.opt.api import (
+    CostFunction,
+    Optimizer,
+    construct_history_info,
+    optimization_result,
+)
+from orquestra.opt.history.recorder import RecorderFactory, recorder as _recorder
+from orquestra.opt.optimizers.pso.topologies import StarTopology, SwarmTopology
+
+Bounds = Union[
+    scipy.optimize.Bounds, Sequence[Tuple[float, float]], Tuple[float, float]
+]
+
+
+def _get_bounds_like_array(
+    bounds: Bounds,
+) -> Union[np.ndarray, Tuple[np.ndarray, np.ndarray]]:
+    """
+    Casts a Bounds object to an object with two entries: the first being the lower
+    bounds and the second being the upper bounds.
+
+    Parameters
+    ----------
+    bounds : Bounds
+        A Bounds object, which can be a scipy.optimize.Bounds object, a sequence of
+        tuples, one tuple being the bonds per parameter, or a tuple of two floats,
+        which are the lower and upper bounds for all parameters.
+
+    Returns
+    -------
+    Union[np.ndarray, Tuple[np.ndarray, np.ndarray]]
+        Lower and upper bounds.
+    """
+    if isinstance(bounds, scipy.optimize.Bounds):
+        return bounds.lb, bounds.ub
+    else:
+        return np.array(bounds).T
+
+
+class PSOOptimizer(Optimizer):
+    def __init__(
+        self,
+        swarm_size: int,
+        bounds: Bounds,
+        inertia: float,
+        affinity_towards_best_particle_position: float,
+        affinity_towards_best_swarm_position: float,
+        patience: Optional[int] = None,
+        delta: float = 1e-10,
+        max_iterations: Optional[int] = None,
+        max_fevals: Optional[int] = None,
+        learning_rate: float = 1.0,
+        velocity_bounds: Optional[Bounds] = None,
+        topology_constructor: Callable[[int], SwarmTopology] = StarTopology,
+        seed: Optional[int] = None,
+        recorder: RecorderFactory = _recorder,
+    ):
+        """
+        Constructor of the Particle Swarm Optimizer for continuous variables.
+        When you call `.minimize`, the initial positions of the particles can be
+        set using the method `get_initial_params`, which by default samples
+        uniformly from the parameter space.
+
+        Parameters
+        ----------
+        swarm_size : int
```

Comment:
[comment] No need to include type here, as the type is already in the signature. When you have them in two places, you always need to remember to update them in both, and as we know, humans make mistakes.

Reply:
Does sphinx render the type automatically from the signature?

Reply:
Well, I'm not entirely sure.
If not, then we need to refactor our whole codebase anyway 🙃 
I just checked in our docs and I'm not sure what I'm seeing there, so it requires some extra investigation.

Reply:
Ok, could you change the style of the docs to be consistent with how it's being formatted in other places?

Reply:
Actually, I had already eliminated the types in the docstrings as well! So this has been dealt with 😄

Reply:
Thank you, merging it then!